src/actions/SaveSubmission.js

```diff
@@ -204,16 +204,21 @@ module.exports = function(router) {
 
         if (this.settings.transform) {
           try {
-            const newData = (new VM({
+            let vm = new VM({
               timeout: 500,
-              sandbox: _.cloneDeep({
-                submission: (res.resource && res.resource.item) ? res.resource.item : req.body,
-                data: submission.data,
-              }),
+              sandbox: {},
               eval: false,
               fixAsync: true
-            })).run(this.settings.transform);
+            });
+
+            const vmSubmission = (res.resource && res.resource.item) ? res.resource.item : req.body;
+
+            vm.freeze(vmSubmission, 'submission');
```

Comment:
There is no need to freeze these. They are allowed to mutate their own submission and data objects.